client.go

```diff
@@ -265,37 +253,192 @@ func (c *Client) CreateInstance(Name string, CPUs int, Memory int,
 	return instance, nil
 }
 
+// snapshotRequest defines options when making a snapshot of an instance
+type snapshotRequest struct {
+	All bool `json:"all"`
+}
+
+// SnapshotInstance takes a snapshot of an instance
+func (c Client) SnapshotInstance(instanceUUID string, all bool) error {
+	path := fmt.Sprintf("instances/%s/snapshot", instanceUUID)
+
+	request := &snapshotRequest{
+		All: all,
+	}
+	post, err := json.Marshal(request)
+	if err != nil {
+		return err
+	}
+
+	_, err = c.httpRequest(path, "POST", *bytes.NewBuffer(post))
+
+	return err
+}
+
+// Snapshot defines a snapshot of an instance
+type Snapshot struct {
+	UUID    string `json:"uuid"`
+	Device  string `json:"device"`
+	Created string `json:"created"` // TODO: Change to time.Time
+}
+
+// GetInstanceSnapshots fetches a list of instance snapshots
+func (c *Client) GetInstanceSnapshots(instanceUUID string) ([]Snapshot, error) {
+	path := fmt.Sprintf("instances/%s/snapshot", instanceUUID)
+	body, err := c.httpRequest(path, "GET", bytes.Buffer{})
+	if err != nil {
+		return []Snapshot{}, err
+	}
+
+	snapshots := []Snapshot{}
+	err = json.NewDecoder(body).Decode(&snapshots)
+	if err != nil {
+		return []Snapshot{}, err
+	}
+	return snapshots, nil
+}
+
+// RebootInstance reboots an instance
+func (c Client) RebootInstance(instanceUUID string) error {
+	path := fmt.Sprintf("instances/%s/reboot", instanceUUID)
+	_, err := c.httpRequest(path, "POST", bytes.Buffer{})
+	return err
+}
+
+// PowerOffInstance powers on an instance
+func (c Client) PowerOffInstance(instanceUUID string) error {
+	path := fmt.Sprintf("instances/%s/poweroff", instanceUUID)
+	_, err := c.httpRequest(path, "POST", bytes.Buffer{})
+	return err
+}
+
+// PowerOnInstance powers on an instance
+func (c Client) PowerOnInstance(instanceUUID string) error {
+	path := fmt.Sprintf("instances/%s/poweron", instanceUUID)
+	_, err := c.httpRequest(path, "POST", bytes.Buffer{})
+	return err
+}
+
+// PauseInstance will pause an instance
+func (c Client) PauseInstance(instanceUUID string) error {
+	path := fmt.Sprintf("instances/%s/pause", instanceUUID)
+	_, err := c.httpRequest(path, "POST", bytes.Buffer{})
+	return err
+}
+
+// UnPauseInstance will unpause an instance
+func (c Client) UnPauseInstance(instanceUUID string) error {
+	path := fmt.Sprintf("instances/%s/unpause", instanceUUID)
+	_, err := c.httpRequest(path, "POST", bytes.Buffer{})
+	return err
+}
```

Comment:
Do you think that all this repetitive code should be done more like:

```
func (c *Client) thingInstance(instanceUUID string, thing string) error {
    path := fmt.Sprintf("instances/%s/%s", instanceUUID, thing)
    _, err := c.httpRequest(path, "POST", bytesBuffer{})
    return err
}

func (c *Client) UnPauseInstance(instanceUUID string) error {
    return thingInstance(instanceUUID, "unpause")
}
```

If so, is there a nice way we could handle the case where its not the URL which changes, but the HTTP verb instead?